### ISSUE TITLE
Unable to push workspace to self-hosted gitlab

### ISSUE THREAD

**Issue by kreosus** [getinsomnia/insomnia#2240]
**Describe the bug**
User is unable to Unable to push workspace to  self-hosted gitlab

**To Reproduce**
User was able to put together 2 endpoints and has setup a gitlab repo and got an access token for Kong Studio. However when he pushes the code, he will get a 401 Unauthorized message

**Expected behavior**
A connection to GitLab
**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows
 - Installation Method: Release page
 - App Version v2020.2.0

![image002](https://user-images.githubusercontent.com/59621250/83576215-d921c300-a4f6-11ea-91fd-912a92ac780e.jpg)

![image005](https://user-images.githubusercontent.com/59621250/83576226-de7f0d80-a4f6-11ea-804d-df43a2b31b4f.jpg)

![image006](https://user-images.githubusercontent.com/59621250/83576238-e5a61b80-a4f6-11ea-93c5-6325e0afcae2.jpg)

**Additional context**

**Comment by develohpanda** [comment]
Possibly related to #2221, which fails with Bitbucket Server (also a self-hosted instance).

It is possible that these tokens for self-hosted instances need to be sent as bearer tokens instead of as basic auth (looking at the bitbucket server documentation linked in #2221).

isomorphic-git has the ability to send [explicit headers](https://isomorphic-git.org/docs/en/onAuth#option-2-headers), or even use the [native credential manager](https://isomorphic-git.org/docs/en/snippets#use-native-git-credential-manager).

**Comment by develohpanda** [comment]
This testing is using gitlab-ce 11.10.8, self-hosted on Azure.

> Note: This version does not have the `write_repository` scope available, which is introduced in gitlab-ce 11.11.

# tl;dr
1. Granting the `api` scope to the access token will allow for clone, push and pull of all branches.
2. Trying to push to a protected branch, when a user does not have permission to push to that branch, **silently** fails. Insomnia does not prompt of any failure to push to a protected branch, but the commits never end up in the repository. Needs further investigation working with protected branches on other VCS providers too. #2371 
3. I was unable to reproduce 401 on push, as reported by this original issue
4. After cloning, changing branches in Insomnia _didn't_ work at all, I saw no updates in the UI. #2365

## User as developer
* Not an admin
* User has “developer” permission on project
* 2FA enabled

### With password
* Clone: 🕵️‍♀️ 401: Unauthorized

### With access token containing  `read_repository`  permission
* Clone: ✅
* Push master: 🚫 403: Forbidden
* Push branch: 🚫 403: Forbidden
* Pull: ✅

### With access token containing `api`  permission
* Clone: ✅
* Push master: 🐛  
	* **Success on the UI, silently fails to push in the background**
	* By default `master` is a protected branch in gitlab, and those with a `developer` role cannot push to it. Insomnia should inform of the failure.
* Push branch: ✅
* Pull: ✅

- - - -

## User as maintainer
* Not an admin
* User has “maintainer” permission on project
* 2FA enabled

### With password
* Clone: 🕵️‍♀️ 401: Unauthorized

### With access token containing  `read_repository`  permission
* Clone: ✅ 
* Push: 🚫 403: Forbidden
* Pull: ✅ 

### With access token containing `api`  permission
* Clone: ✅
* Push: ✅
* Pull: ✅

----

![image](https://user-images.githubusercontent.com/4312346/86203764-e98e9180-bbb9-11ea-875b-ba04594b1809.png)
